File: Sakila_Net/Sakila_Net/VB_SakilaLib/Model/Country.vb
Namespace Cl
  Namespace Harrys
    Namespace Sakila
       Namespace Model
public class ClCountry

   private countryId as Integer
   private country as String
   private lastUpdate as Date


   public Property DataCountryId() as Integer
        Get
            Return countryId
        End Get
        Set(ByVal Value as Integer)
            countryId = Value
        End Set        
   End Property   
   public Property DataCountry() as String
        Get
            Return country
        End Get
        Set(ByVal Value as String)
            country = Value
        End Set        
   End Property   
   public Property DataLastUpdate() as Date
        Get
            Return lastUpdate
        End Get
        Set(ByVal Value as Date)
            lastUpdate = Value
        End Set        
   End Property   

   Public Sub New (countryId    as Integer,country    as String,lastUpdate    as Date)
		   dataCountryId = countryId
		   dataCountry = country
		   dataLastUpdate = lastUpdate
   End Sub
   Public Sub New (countryId    as Integer)
    dataCountryId = countryId
   End Sub


   public Overrides Function  ToString() As String
        '// Registro JSon
		return "{" _ 
		    & "," & chr(34) & "countryId" & chr(34) & ":" & chr(34) & dataCountryId & chr(34) _
		    & "," & chr(34) & "country" & chr(34) & ":" & chr(34) & dataCountry & chr(34) _
		    & "," & chr(34) & "lastUpdate" & chr(34) & ":" & chr(34) & dataLastUpdate & chr(34) _
		 & "}"
   End Function   
   public Sub imprimir()
        Console.WriteLine(ToString())
   End Sub
   public Function jsonSerialize()
           return "get_object_vars($this);"
   End Function   
End Class
         End Namespace  
      End Namespace  
   End Namespace  
End Namespace  

File: Sakila_Php/Entities/Country.php
<?PHP
class Country implements JsonSerializable{

   private $countryId;
   private $country;
   private $lastUpdate;


   public function getCountryId(){
        return $this->countryId;
   }
   public function setCountryId($countryId){
        $this->countryId = $countryId;
   }
   
   public function getCountry(){
        return $this->country;
   }
   public function setCountry($country){
        $this->country = $country;
   }
   
   public function getLastUpdate(){
        return $this->lastUpdate;
   }
   public function setLastUpdate($lastUpdate){
        $this->lastUpdate = $lastUpdate;
   }
   

   public function Country($countryId,$country='',$lastUpdate=''){
		   $this->setCountryId($countryId);
		   $this->setCountry($country);
		   $this->setLastUpdate($lastUpdate);
   }
   public function __constructXX($countryId,$country='',$lastUpdate='') {
		   $this->setCountryId($countryId);
		   $this->setCountry($country);
		   $this->setLastUpdate($lastUpdate);
   }
   public function __toString(){
        // Registro JSon
		return "{" 
		    . "," . chr(34) . "countryId" . chr(34) . ":" . chr(34) . $this->getCountryId() . chr(34) 
		    . "," . chr(34) . "country" . chr(34) . ":" . chr(34) . $this->getCountry() . chr(34) 
		    . "," . chr(34) . "lastUpdate" . chr(34) . ":" . chr(34) . $this->getLastUpdate() . chr(34) 
		 . "}";
   }   
   public function imprimir()
   {
        echo $this->__toString();
   }
   public function jsonSerialize()
   {
        return get_object_vars($this);
   }   
}
// Para realizar pruebas
//$actor = new Country();
//echo "Imprimiendo ";$actor->imprimir();
//echo "ToString : " . $actor;
?>
